### PR TITLE
Greying out pending nodes in the IDE

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Adding node `Kind::Pending` and associating it with some _grey like_ color schemes.

### Important Notes

Some of us might have been hoping for a fix completely on the engine side, but it is already a month and the fix doesn't exist yet. It is not clear how much work is required to create such fix. Rather than hoping for it, I propose to use the `Pending` information which is already available in the IDE since #3729 

I don't claim to be a visual expert. I am ready to adapt the visual presentation according to suggestions of the IDE guys.

### Checklist

Please include the following checklist in your PR:

- [x] All code conforms to the
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
Here is a video [showing the greying out](https://user-images.githubusercontent.com/26887752/214882106-af3a3737-5eb9-4e0d-b40f-5f1cda8f84a2.mp4) behavior. The source code of the example is available in [pivotal](https://www.pivotaltracker.com/story/show/182818735/comments/233402238).

**Review by Frizi — APPROVED** [review]
From code point of view, all looks good. Visually I'm not sure if using the stripes here is right, as it calls for attention a lot. Maybe using a little less strong gray color would make it look better as is.

File: app/gui/src/presenter/graph/state.rs [hunk 1]
```diff
@@ -532,7 +532,7 @@ impl<'a> ControllerChange<'a> {
             None | Some(Value) => None,
             Some(DataflowError { trace }) => Some((Kind::Dataflow, None, trace)),
             Some(Panic { message, trace }) => Some((Kind::Panic, Some(message), trace)),
-            Some(Pending { .. }) => None,
+            Some(Pending { .. }) => Some((Kind::Pending, Some("Pending...".to_string()), Vec::new())),
```
Comment (wdanilo) [inline app/gui/src/presenter/graph/state.rs]: extract literals to consts, please.

**Review by wdanilo — CHANGES_REQUESTED** [review]
This is amazing, however, can we have video attached to this PR showing thi in action and showing corner cases, to be sure this PR works as intended? :)

**Comment by JaroslavTulach** [comment]
> ... can we have video attached to this PR showing this in action and showing corner cases, to be sure this PR works as intended? :)

I had some problems uploading the video and embedding it, but here is a [link to mp4 file](https://user-images.githubusercontent.com/26887752/214882106-af3a3737-5eb9-4e0d-b40f-5f1cda8f84a2.mp4) - I am able to successfully play it on Chrome (but not on Firefox) or download it and play it with `vlc`.

File: app/gui/src/presenter/graph/state.rs (hunk 1 above)
Comment (JaroslavTulach) [reply app/gui/src/presenter/graph/state.rs]: Resolved in [bca0fbb](https://github.com/enso-org/enso/pull/4088/commits/bca0fbb8162d35cc201dc0dd9ed0bc976dea3e0a)

**Comment by JaroslavTulach** [comment]
> From code point of view, all looks good. 

Thanks a lot Pawel for helping me navigate in the IDE codebase!

> Maybe using a little less strong gray color would make it look better as is.

Anyone with more visual design skills than me (e.g. everyone), feel free to suggest changes or directly commit into this PR and improve colors & etc.

**Review by wdanilo — CHANGES_REQUESTED** [review]
Based on the video I see two big issues:
1. The graph turns red when editing the node. It should not.
2. When nodes are marked as not computed, the optional arguments are not displayed. We still know what these nodes refer to, so the arguments should be there.

**Comment by JaroslavTulach** [comment]
> The graph turns red when editing the node. It should not.

That behavior not related to my changes. If you think that is a bug, it needs to be addressed separately. And yes, I find the _"turning red"_ behavior strange as well.

As discussed with Wojciech I've reported this problem in [pivotal](https://www.pivotaltracker.com/n/projects/2539304/stories/184347990).

**Comment by JaroslavTulach** [comment]
> When nodes are marked as not computed, the optional arguments are not displayed. We still know what these nodes refer to, so the arguments should be there.

I'd like to fix that, but that is way deeper in the IDE code than I can dig into easily. Btw. the [video associated](https://user-images.githubusercontent.com/357683/215092755-0010e41d-a2cf-447a-900e-4619408effa0.webm) with #4006 shows the same issue - IDE doesn't render _optional arguments_ when node is in _pending/errorneous/non-normal_ state.

**Comment by JaroslavTulach** [comment]
At the end #4006 got integrated quickly and there is no need for this PR anymore. Thanks everyone who provided me guidance, reviews and pointed out visual flaws in my proposed implementation.

**Pull request closed without merging**